[ENG-8954][docs] EAS Update: add pitch and revamp intro page (#23095)
expo/expo: docs/pages/eas-update/introduction.mdx

Still using Classic Updates? We moved those docs to [the archive](/archive/classic-updates/introduction).

## Pitch

> #### Improve User Experience

Reviewer: ```suggestion
### Improve User Experience
```

Let's not skip heading levels (h2 to h4) as per our writing style guide: https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md#headings

> With EAS Update, you can release hotfixes and enhancements in response to user feedback or market trends, keeping your users engaged and satisfied with your app. Don't wait for the next planned update or manual update request.

Reviewer: ```suggestion
EAS Update allows you to release hotfixes and enhancements in response to user feedback or market trends. It helps you to keep your users engaged and satisfied with your app. Don't wait for the next planned update or manual update request.
```

> #### Broad Compatibility

Reviewer: ```suggestion
### Broad Compatibility
```

> EAS Update works with vanilla React Native apps, all you need to do is install the lightweight `expo` npm package as a prerequisite, making it a compelling alternative to solutions like [Codepush](/eas-update/codepush).

Reviewer: ```suggestion
EAS Update works with vanilla React Native apps. All you need to do is install the lightweight `expo` npm package as a prerequisite, making it a compelling alternative to solutions such as [Codepush](/eas-update/codepush).
```

Instead of adding a pause, we can split it in two sentences.

> #### Tailored for Expo

Reviewer: ```suggestion
### Tailored for Expo
```

EAS Update is designed for Expo apps, making integration easy and workflows efficient. Use the website UIs to check app status and debug, especially for developers using EAS Build.

> #### Customizable Update Strategies

Reviewer: ```suggestion
### Customizable Update Strategies
```

> Apply updates through our [expo-updates API](/versions/latest/sdk/updates) and [app.json configuration](/versions/latest/config/app/#updates) if the default behavior is not suitable for you. You’ll always have the upper hand in shaping the update process without compromising your users' experience.

Reviewer: ```suggestion
Apply updates through [`expo-updates` API](/versions/latest/sdk/updates) and [app config configuration](/versions/latest/config/app/#updates) if the default behavior is not suitable for you. You’ll always have the upper hand in shaping the update process without compromising your users' experience.
```

- Since `expo-updates` is a library
